### DISCUSSION TITLE
YOLO11 validation without augmentation.

### DISCUSSION THREAD

**Discussion by NS-Nik** [ultralytics/ultralytics#17658] [Q&A] [answered]
Hi! 

I have trained yolo11m.pt for my custom objects but I'm facing some issues during validation. I implement the aggressive augmentation settings  during the model training. However, I don't want to use it for the calculation of the validation metrics.  I tried to switch it off with the 'augment=False' setting. But the augmentation  is still  implemented. 

`metrics = model.val(augment=False, iou=0.25)`

Any ideas  how to turn of the augmentation for the validation?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @NS-Nik, thank you for reaching out to the Ultralytics community 🚀! Your issue with disabling augmentation during validation is a valuable topic. We invite you to explore our [Docs](https://docs.ultralytics.com) where you might find insight into validation settings. Specifically, our [Python Usage](https://docs.ultralytics.com/usage/python/) guide could be of assistance for similar issues.

If this is a 🐛 Bug Report, please help us by providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/), which will assist in debugging the situation more efficiently.

Moreover, if you have custom training ❓ questions, please share detailed information, including dataset samples and training logs. Double-check to ensure you're following the guidelines in our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To engage with the community, consider joining our [Discord](https://ultralytics.com/discord) 🎧 for real-time interactions, or for deeper dives, tap into our [Discourse](https://community.ultralytics.com) or our [Subreddit](https://reddit.com/r/ultralytics) to fellowship with other members. 

## Upgrade

To ensure you’re running the latest version, upgrade the `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

Your YOLO model can operate in verified environments offering a range of dependencies:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** via Deep Learning VM. Visit [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** via Deep Learning AMI. Visit [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: Check the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

This badge displays the status of [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule), which confirms the operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across multiple OS environments.

Don't worry, this is an automated response 🤖, but rest assured, an Ultralytics engineer will be here to assist you further soon!

**Comment by glenn-jocher** [comment]
@NS-Nik hi! To ensure no augmentation is applied during validation, confirm that the validation function is correctly configured without any augmentation settings in your YAML or script. Also, check that there are no residual augmentation parameters overriding your `augment=False` setting. If the issue persists, try updating to the latest version of the Ultralytics package, as there may have been fixes or improvements. For further assistance, feel free to review the [YOLO Common Issues](https://docs.ultralytics.com/guides/yolo-common-issues/) documentation.

**Comment by NS-Nik** [comment]
HI! @glenn-jocher 
I created the .yaml file where I defined  paths to my validation images and the passed it to the "data" argument and it worked! 

Thank you for the help!

**Reply by glenn-jocher** [reply]
@NS-Nik i'm glad to hear that it worked for you! If you have any more questions or need further assistance, feel free to reach out. Kudos to the YOLO community and the Ultralytics team for their support!

**Reply by Skillnoob** [reply, marked as answer]
Augmentation isn't used during validation and the `augment` parameter is only for prediction, to answer your other points.
